### PR TITLE
refactor: update Python test input to accept Option<&str>

### PR DESCRIPTION
**Pull request by AvivYossef-starkware**
- Changed Python test input type from `&str` to `Option<&str>`.
- This modification allows flexibility by handling cases where the input might not be provided.
- Adjusted the function signatures and updated relevant documentation.

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)

### CODE REVIEW

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)* status: all files reviewed, 3 unresolved discussions (waiting on @AvivYossef-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNqkaDjeAaH6_wOWb:-NxMNqkaDjeAaH6_wOWc:b60jgnd) (no related file):*
python side?

___
___
*[`crates/committer_cli/src/main.rs` line 46 at r1](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMN30s2hI_Jjg2ig7q:-NxMN30s2hI_Jjg2ig7r:b-d8falr) ([raw file](https://github.com/starkware-libs/committer/blob/aaa1a6b3b4f45366c71e60442037294efcfe7ceb/crates/committer_cli/src/main.rs#L46)):*
> ```XML
>         /// Test inputs as a json.
>         #[clap(long)]
>         inputs: Option<String>,
> ```

so now if `--inputs` doesn't exist in the command, this will be `None`?



_Code quote:_
```XML
        #[clap(long)]
        inputs: Option<String>,
```
___
*[`crates/committer_cli/src/tests/python_tests.rs` line 95 at r1](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNU1zEvCux0bUHrGb:-NxMNU1zEvCux0bUHrGc:undefined) ([raw file](https://github.com/starkware-libs/committer/blob/aaa1a6b3b4f45366c71e60442037294efcfe7ceb/crates/committer_cli/src/tests/python_tests.rs#L95)):*
> ```XML
>                     serde_json::from_str(input.ok_or(PythonTestError::NoneInputError)?)?;
>                 Ok(test_binary_serialize_test(binary_input))
>             }
> ```





_Suggestion:_
```XML
impl PythonTest {
    fn non_optional_input(input: Option<&str>) -> Result<String, PythonTestError> {
        input.ok_or(PythonTestError::NoneInputError)
    }
    
    /// Runs the test with the given arguments.
    pub(crate) fn run(&self, input: Option<&str>) -> Result<String, PythonTestError> {
        match self {
            Self::ExampleTest => {
                let example_input: HashMap<String, String> =
                    serde_json::from_str(Self::non_optional_input(input)?)?;
                Ok(example_test(example_input))
            }
            Self::FeltSerialize => {
                let felt = Self::non_optional_input(input)?
                    .parse::<u128>()?;
                Ok(felt_serialize_test(felt))
            }
            Self::HashFunction => {
                let hash_input: HashMap<String, u128> =
                    serde_json::from_str(Self::non_optional_input(input)?)?;
                Ok(test_hash_function(hash_input))
            }
            Self::BinarySerialize => {
                let binary_input: HashMap<String, u128> =
                    serde_json::from_str(Self::non_optional_input(input)?)?;
                Ok(test_binary_serialize_test(binary_input))
            }
```

**Review by AvivYossef-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)* status: 1 of 2 files reviewed, 3 unresolved discussions (waiting on @dorimedini-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNqkaDjeAaH6_wOWb:-NxMPFKY3mhyRfmKFc71:bfpjor4) (no related file):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

python side?
</blockquote></details>

You asked me here https://reviewable.io/reviews/starkware-industries/starkware/34900#- to refactor inputs to an optional argument. 

___
___
*[`crates/committer_cli/src/main.rs` line 46 at r1](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMN30s2hI_Jjg2ig7q:-NxMPZt84fFzFVeS728B:bb77ufg) ([raw file](https://github.com/starkware-libs/committer/blob/aaa1a6b3b4f45366c71e60442037294efcfe7ceb/crates/committer_cli/src/main.rs#L46)):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

so now if `--inputs` doesn't exist in the command, this will be `None`?
</blockquote></details>

Yes.
___
*[`crates/committer_cli/src/tests/python_tests.rs` line 95 at r1](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNU1zEvCux0bUHrGb:-NxMSdqp24_JMKF0g7NE:b-896fix) ([raw file](https://github.com/starkware-libs/committer/blob/aaa1a6b3b4f45366c71e60442037294efcfe7ceb/crates/committer_cli/src/tests/python_tests.rs#L95)):*
> ```XML
>                     serde_json::from_str(input.ok_or(PythonTestError::NoneInputError)?)?;
>                 Ok(test_binary_serialize_test(binary_input))
>             }
> ```

Done.

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)* status: all files reviewed, 1 unresolved discussion (waiting on @AvivYossef-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNqkaDjeAaH6_wOWb:-NxMmiBZACF-7bTm8Smb:btkl05b) (no related file):*
<details><summary><i>Previously, AvivYossef-starkware wrote…</i></summary><blockquote>

You asked me here https://reviewable.io/reviews/starkware-industries/starkware/34900#- to refactor inputs to an optional argument.
</blockquote></details>

yes but is PR 34900 pointing to this committer version?
if not, please update it

___

**Review by AvivYossef-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)* status: all files reviewed, 1 unresolved discussion (waiting on @dorimedini-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNqkaDjeAaH6_wOWb:-NxMncdNBoPO1-IFCYEs:batg64f) (no related file):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

yes but is PR 34900 pointing to this committer version?
if not, please update it
</blockquote></details>

I'm waiting for this commit to merge, since the commit hash is changed after the merge.

___

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)* status: all files reviewed, 1 unresolved discussion (waiting on @AvivYossef-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNqkaDjeAaH6_wOWb:-NxMp-AP1rO3f6OuuW2T:b-a74fps) (no related file):*
<details><summary><i>Previously, AvivYossef-starkware wrote…</i></summary><blockquote>

I'm waiting for this commit to merge, since the commit hash is changed after the merge.
</blockquote></details>

but first you need to test the python-side CI.
since this PR has an effect on the python side, you can't merge until a python-side CI run passes with this commit (`49a5ce4` ATM)

___

**Review by dorimedini-starkware — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)* status: all files reviewed, 1 unresolved discussion (waiting on @AvivYossef-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNqkaDjeAaH6_wOWb:-NxN1nJd6tNLr5qm6iTH:b-5x9asu) (no related file):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

but first you need to test the python-side CI.
since this PR has an effect on the python side, you can't merge until a python-side CI run passes with this commit (`49a5ce4` ATM)
</blockquote></details>

ok, I see the commit is updated; LMK when the python CI passes and I'll unblock

___

**Review by AvivYossef-starkware — COMMENTED** [review]
> Reviewed 1 of 2 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)* status: all files reviewed, 1 unresolved discussion (waiting on @AvivYossef-starkware)

**Review by AvivYossef-starkware — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102)* status: all files reviewed, 1 unresolved discussion (waiting on @dorimedini-starkware)

*[a discussion](https://reviewable.io/reviews/starkware-libs/committer/102#-NxMNqkaDjeAaH6_wOWb:-NxO7I7a24o45FW6CBrI:b-896fix) (no related file):*
<details><summary><i>Previously, dorimedini-starkware wrote…</i></summary><blockquote>

ok, I see the commit is updated; LMK when the python CI passes and I'll unblock
</blockquote></details>

Done.

___

**Review by dorimedini-starkware — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/starkware-libs/committer/102#-:-NxOuZzPAHD6kGekOJE7:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @AvivYossef-starkware)

**Pull request merged**
